projects/Apigen/apigen/Apigen.cs

```diff
@@ -1108,10 +1108,14 @@ public void MaybeEmitModelMethod(MethodInfo method)
 
         public string SanitisedFullName(Type t)
         {
-            if (t.FullName.StartsWith("System.Collections.Generic.IDictionary`2[[System.String"))
+            if (t.Equals(typeof(IDictionary<string, object>)))
```

Comment:
The previous approach had a potential bug in that it would treat any `IDictionary<string, T>` as `IDictionary<string, object>`, so this is definitely better.